ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,35 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  try
+    match List.assoc "secureboot" platform with
+    | "auto" -> ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) :: (List.remove_assoc "secureboot" platform)
+    | _ -> platform
+  with Not_found -> platform
```

Comment:
This can also be done by using `| exception Not_found -> platform` instead of wrapping with try/with but that is a recent language feature, we can keep this as is.

Reply:
It's tidier, and I think looks better. I'd like to change it